### PR TITLE
feat(web-components): add SSR support via Declarative Shadow DOM modules

### PR DESCRIPTION
**Pull request by radium-v**
## Previous Behavior

`@fluentui/web-components` had no server-side rendering path. Components were defined and rendered only on the client, so SSR frameworks could not pre-render Fluent web components into HTML and hydrate them on load.

The e2e tests ran against a custom harness that only covered client-side rendering.

## New Behavior

`@fluentui/web-components` now supports SSR via Declarative Shadow DOM, built on `@microsoft/fast-element` and `@microsoft/fast-html`. Every component in the package ships a paired `define-async.ts` / `*.definition-async.ts` module so its template can render on the server and hydrate on the client.

The build also emits two new asset types alongside the JS modules, one pair per component (42 of each), exposed through new package subpath exports:

- `@fluentui/web-components/<component>/template.html`: declarative HTML template wrapped in `<f-template>`, consumed by `@microsoft/fast-html` to render Declarative Shadow DOM on the server.
- `@fluentui/web-components/<component>/styles.css`: plain CSS extracted from the component's styles, importable as a raw stylesheet without pulling in the component's JS.

Currently, the SSR templates and styles are generated at build time and placed in the `dist` output, but we may want to provide them as src assets in the future for easier consumption and customization.

New test harness modules (`test/src/entry-client.ts`, `test/src/entry-server.ts`) and a `test/ssr.html` entry point template exercise the DSD output end-to-end. The e2e pipeline runs SSR tests after CSR tests.

Several components needed template or base adjustments to render correctly under SSR: tooltip (anchor positioning polyfill handling), text-input, textarea, tree, and tree-item. The `theme/set-theme` tests are skipped under SSR.

## Next Steps

This PR lays the groundwork for SSR support in `@fluentui/web-components`, but there are still some follow-up tasks to complete:

- Update to the latest `@microsoft/fast-element` and remove the `@microsoft/fast-html` dependency, as DSD rendering will be directly supported in FAST in the next major release.
- Provide more detailed documentation and examples for using the SSR templates and styles.
- Add more comprehensive SSR tests covering edge cases and hydration scenarios.

### CODE REVIEW

File: packages/web-components/src/accordion/accordion.definition-async.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { type PartialFASTElementDefinition } from '@microsoft/fast-element';
+import { tagName } from './accordion.options.js';
+
+export const declarativeDefinition: PartialFASTElementDefinition = {
```
Comment (radium-v) [inline packages/web-components/src/accordion/accordion.definition-async.ts:12]: missing doc block

File: packages/web-components/src/accordion-item/define-async.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { RenderableFASTElement } from '@microsoft/fast-html';
+import { AccordionItem } from './accordion-item.js';
+import { definition } from './accordion-item.definition-async.js';
```
Comment (radium-v) [inline packages/web-components/src/accordion-item/define-async.ts]: import ordering

File: packages/web-components/src/dropdown/dropdown.base.ts [hunk 3]
```diff
@@ -448,7 +444,12 @@ export class BaseDropdown extends FASTElement {
    * @public
    */
   public get enabledOptions(): DropdownOption[] {
-    return this.listbox?.enabledOptions ?? [];
+    return (
+      this.listbox?.enabledOptions ??
+      Array.from(this.querySelectorAll('*')).filter<DropdownOption>(
+        (o): o is DropdownOption => isDropdownOption(o) && !o.disabled,
+      )
```
Comment (radium-v) [inline packages/web-components/src/dropdown/dropdown.base.ts]: The filtering approach should be normalized between this and line 518

File: packages/web-components/src/menu/define-async.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { RenderableFASTElement } from '@microsoft/fast-html';
+import { Menu } from './menu.js';
+import { definition } from './menu.definition-async.js';
```
Comment (radium-v) [inline packages/web-components/src/menu/define-async.ts]: import order

File: packages/web-components/src/textarea/textarea.spec.ts [hunk 5]
```diff
@@ -151,38 +151,31 @@ test.describe('TextArea', () => {
...
-          const root = node.shadowRoot?.querySelector('.root');
-          return root?.contains(node.autoSizerEl ?? null) ?? false;
-        });
+      await page.addInitScript({
+        content: `
+          const originalSupports = CSS.supports.bind(CSS);
+          window.__originalCSSSupports = originalSupports;
+          CSS.supports = (property, value) => {
+            if (property === 'field-sizing: content' || (property === 'field-sizing' && value === 'content')) {
+              console.log('Mocking CSS.supports to return false for', property);
```
Comment (radium-v) [inline packages/web-components/src/textarea/textarea.spec.ts]: console log

File: packages/web-components/src/tooltip/tooltip.ts [hunk 6]
```diff
@@ -85,6 +83,8 @@ export class Tooltip extends FASTElement {
   public connectedCallback(): void {
     super.connectedCallback();
 
+    this.popover ??= 'auto';
```
Comment (radium-v) [inline packages/web-components/src/tooltip/tooltip.ts:86]:
```suggestion
    this.popover ||= 'auto';
```

File: packages/web-components/src/tree-item/tree-item.base.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { attr, css, type ElementStyles, FASTElement, observable } from '@microsoft/fast-element';
+import { attr, css, type ElementStyles, FASTElement, observable, Updates } from '@microsoft/fast-element';
```
Comment (radium-v) [inline packages/web-components/src/tree-item/tree-item.base.ts]: unused import

**Review by tudorpopams — APPROVED** [review]

File: packages/web-components/src/tooltip/tooltip.ts (hunk 6 above)
Comment (radium-v) [reply packages/web-components/src/tooltip/tooltip.ts:86]: `popover` is typed `string | null` so this is already correct

File: packages/web-components/project.json [hunk 8]
```diff
@@ -3,5 +3,10 @@
   "$schema": "../../node_modules/nx/schemas/project-schema.json",
   "projectType": "library",
   "implicitDependencies": [],
-  "tags": ["platform:web", "web-components"]
+  "tags": ["platform:web", "web-components"],
+  "targets": {
+    "e2e": {
+      "dependsOn": ["build"]
+    }
+  }
```
Comment (radium-v) [inline packages/web-components/project.json:11]: This is addressed in #36257

File: packages/web-components/src/menu/menu.template.ts [hunk 9]
```diff
@@ -3,13 +3,7 @@ import type { Menu } from './menu.js';
 
 export function menuTemplate<T extends Menu>(): ElementViewTemplate<T> {
   return html<T>`
-    <template
-      ?open-on-hover="${x => x.openOnHover}"
-      ?open-on-context="${x => x.openOnContext}"
-      ?close-on-scroll="${x => x.closeOnScroll}"
-      ?persist-on-item-click="${x => x.persistOnItemClick}"
-      @keydown="${(x, c) => x.menuKeydownHandler(c.event as KeyboardEvent)}"
-    >
```
Comment (davatron5000) [inline packages/web-components/src/menu/menu.template.ts:12]: Did we lose these optional features? Are we handling the event binding in the class?

File: packages/web-components/src/text-input/text-input.template.ts [hunk 10]
```diff
@@ -41,7 +41,7 @@ export function textInputTemplate<T extends TextInput>(options: TextInputOptions
           size="${x => x.size}"
           spellcheck="${x => x.spellcheck}"
           type="${x => x.type}"
-          value="${x => x.initialValue}"
+          value="${x => x.value}"
```
Comment (davatron5000) [inline packages/web-components/src/text-input/text-input.template.ts:43]: Is this a big API change for downstream consumers? Do we still need `initialValue`?

File: packages/web-components/src/menu/menu.template.ts (hunk 9 above)
Comment (radium-v) [reply packages/web-components/src/menu/menu.template.ts:12]: All of these attributes are decorated with `@attr` in the class, and their presence in the template is redundant.

File: packages/web-components/src/text-input/text-input.template.ts (hunk 10 above)
Comment (radium-v) [reply packages/web-components/src/text-input/text-input.template.ts:43]: The downstream variant's DSD template has had this change in place since DSD was introduced. the host element's `value` property is bound with `@attr({ attribute: 'value', mode: 'fromView' })`, so the direct functionality via the host's API isn't affected.

The main discrepancy is that the internal `<input>` will now update its `value` _attribute_ when the host's `value` _property_ changes, but as long as the host's `value` is respected as the definitive _initial value_ (it still sets `initialValue` since the attribute is in `fromView` mode), there shouldn't be any problems with form resets (all of that's still baked into the base class, which uses the `initialValue` and is tied to a dirty-state flag).

Either way, `<fluent-text-input value="something">` is the accepted way to set an initial value, and there's no other way to set the `initialValue` property in a similar declarative fashion.

**Review by davatron5000 — APPROVED** [review]

**Review by marchbox — APPROVED** [review]
